pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -251,6 +251,10 @@ public static void validateIngestionConfig(TableConfig tableConfig, @Nullable Sc
    */
   @VisibleForTesting
   public static void validateIngestionConfig(TableConfig tableConfig, @Nullable Schema schema, boolean disableGroovy) {
+    if (tableConfig.getQueryConfig() != null && tableConfig.getQueryConfig().getDisableGroovyQuery() != null) {
```

Comment:
We should not use the query config to skip the check for ingestion groovy.
After a second thought, we probably should not allow overriding ingestion groovy using table config as the ingestion groovy itself is configured within the table config.

Reply:
So should we just not allow ingestion groovy override then? And only allow groovy override for queries?

Reply:
Removed override for ingestion